src/DocChecks.jl

```diff
@@ -405,4 +408,38 @@ else
     footnotes(doc::Documents.Document) = nothing
 end
 
+
+# Link Checks.
+# ------------
+
+import Requests
+
+function linkcheck(doc::Documents.Document)
+    if doc.user.linkcheck
+        println(" > checking external URLs:")
+        for (src, page) in doc.internal.pages
+            println("   - ", src)
+            for element in page.elements
+                Walkers.walk(page.globals.meta, page.mapping[element]) do block
+                    linkcheck(block, doc)
+                end
+            end
+        end
+    end
+    return nothing
+end
+
+function linkcheck(link::Base.Markdown.Link, doc::Documents.Document)
+    if !(link in doc.internal.locallinks)
+        local good = try
+            Requests.statuscode(Requests.head(link.url)) == 200
+        catch err
+            false
+        end
+        print_with_color(good ? :green : :red, good ? "     " : "   X ", link.url, "\n")
```

Comment:
Also, the perhaps the links could be more indented. My first thought was that the gray page names also were links, but internal ones and therefore got ignored or something:

```
 > checking external URLs:
   - man/contributing.md
   - lib/internals/writers.md
   - man/guide.md
     http://www.mkdocs.org/
     https://github.com/JuliaDocs/Documenter.jl/issues
   - lib/public.md
   - man/examples.md
     https://juliafinance.github.io/Currencies.jl/latest/
   X http://chrisrackauckas.github.io/DifferentialEquations.jl/latest/
```

This I think looks more clear:

```
 > checking external URLs:
   - man/contributing.md
   - lib/internals/writers.md
   - man/guide.md
       http://www.mkdocs.org/
       https://github.com/JuliaDocs/Documenter.jl/issues
   - lib/public.md
   - man/examples.md
       https://juliafinance.github.io/Currencies.jl/latest/
     X http://chrisrackauckas.github.io/DifferentialEquations.jl/latest/
```